### PR TITLE
(DOCSP-23435): Reserved File Names Warning

### PR DESCRIPTION
**Pull request by chloe-ratte**
## Pull Request Info

Added a note/section to the CLI docs landing page to warn for reserved file names users shouldn't use. DRAFT.

### Jira

- https://jira.mongodb.org/browse/DOCSP-23435

### Staged Changes (Requires MongoDB Corp SSO)

- [App Configuration](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-23435/manage-apps/configure/config/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/manage-apps/configure/config.txt [hunk 1]
```diff
@@ -38,6 +38,15 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   From a CLI perspective, there are a few special/reserved file names we advise
+   you to avoid using for your own purposes: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. Files with these names may
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:45]:
```suggestion
   ``realm_config.json``, and ``app_config.json``. Files with these names
```

not may, they _do_ conflict!

File: source/manage-apps/configure/config.txt [hunk 2]
```diff
@@ -38,6 +38,15 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   From a CLI perspective, there are a few special/reserved file names we advise
+   you to avoid using for your own purposes: ``stitch.json``, ``config.json``,
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:44]: can you phrase this more directly? 

you want to convey the following ideas here i think:
* you can include arbitrarily named files in your realm app config
* with the exception of these specific file names, as they are reserved for config files
* including improperly formatted files with these names causes the Realm CLI to throw an error when you import them

File: source/manage-apps/configure/config.txt [hunk 3]
```diff
@@ -38,6 +38,15 @@ using structured JSON configuration and JavaScript source code files.
...
 
+.. important:: Reserved File Names
+   
+   From a CLI perspective, there are a few special/reserved file names we advise
+   you to avoid using for your own purposes: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. Files with these names may
+   conflict with pre-made files of the same name in their app structure's file
+   system. As such, using these names for files may raise CLI errors.
+   
+
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:49]:
```suggestion
```

File: source/manage-apps/configure/config.txt [hunk 4]
```diff
@@ -38,6 +38,14 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   The following file names should not be used: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. Files with these names
+   conflict with pre-made files of the same name in their app structure's file
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:46]: well it's not that they shouldn't be used, it's that these are reserved file names, so they should be used only for these reserved scenarios. 

and the files aren't only 'pre-made'. you can make them following the structure specified in the app configuration reference - https://www.mongodb.com/docs/atlas/app-services/manage-apps/configure/config/

File: source/manage-apps/configure/config.txt [hunk 5]
```diff
@@ -38,6 +38,14 @@ using structured JSON configuration and JavaScript source code files.
...
+   system. Using these file names throws an error when you import the app using the CLI.
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:47]: doesn't always throw error, only if used in a manner differently from the configuration's format.

File: source/manage-apps/configure/config.txt [hunk 6]
```diff
@@ -38,6 +38,16 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   Note the following names: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. These names are reserved specific
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:45]:
```suggestion
   ``realm_config.json``, and ``app_config.json``. These names are reserved for specific 
```

File: source/manage-apps/configure/config.txt [hunk 7]
```diff
@@ -38,6 +38,16 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   Note the following names: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. These names are reserved specific 
+   files used to define and configure specific components in your app. As such, files
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:46]:
```suggestion
   files used to define and configure components in your app. As such, files 
```

File: source/manage-apps/configure/config.txt [hunk 8]
```diff
@@ -38,6 +38,16 @@ using structured JSON configuration and JavaScript source code files.
...
 
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   Note the following names: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. These names are reserved specific 
+   files used to define and configure specific components in your app. As such, files 
+   of these names must conform to a specific schema for its component type. 
+   Using these file names when the file's content fails to conform to that schema
+   will throw an error when you import the app using the CLI.
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:49]:
```suggestion
  throws an error when you import the app using the CLI.
```

File: source/manage-apps/configure/config.txt [hunk 9]
```diff
@@ -38,6 +38,16 @@ using structured JSON configuration and JavaScript source code files.
...
+
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:50]: how about another sentence here w a link to where the user can learn more abt configuring these files? i think this page would be a good place to link to - https://www.mongodb.com/docs/atlas/app-services/manage-apps/configure/config/

Comment (chloe-ratte) [reply source/manage-apps/configure/config.txt:50]: i agree about having a reference, but isn't the link you sent just the same page we're editing now?

File: source/manage-apps/configure/config.txt [hunk 10]
```diff
@@ -38,6 +38,16 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   Note the following names: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. These names are reserved for specific 
+   files used to define and configure components in your App. Files 
+   with these names must conform to a specific schema for its component type.
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:47]:
```suggestion
   The following filenames are used to define and configure components in your App:
   ``stitch.json``, ``config.json``, ``realm_config.json``, and ``app_config.json``.
   Files with these names must conform to a specific schema for its component type. 
```

File: source/manage-apps/configure/config.txt [hunk 11]
```diff
@@ -38,6 +38,16 @@ using structured JSON configuration and JavaScript source code files.
...
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   Note the following names: ``stitch.json``, ``config.json``, 
+   ``realm_config.json``, and ``app_config.json``. These names are reserved for specific 
+   files used to define and configure components in your App. Files 
+   with these names must conform to a specific schema for its component type. 
+   Using these file names when the file's content fails to conform to that schema
+  throws an error when you import the app using the CLI.
+
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:50]:
```suggestion
   
   Refer to this section for reference information about configuring these files.
```

File: source/manage-apps/configure/config.txt [hunk 12]
```diff
@@ -38,6 +38,17 @@ using structured JSON configuration and JavaScript source code files.
 - ``.js`` files define serverless application logic used in functions, triggers,
   HTTPS endpoints, and custom resolvers.
 
+.. important:: Reserved File Names
+   
+   You can include arbitrarily named files in your realm app config, with a few exceptions.
+   The following filenames are used to define and configure components in your App:
+   ``stitch.json``, ``config.json``, ``realm_config.json``, and ``app_config.json``.
+   Files with these names must conform to a specific schema for its component type.
```
Comment (mongodben) [inline source/manage-apps/configure/config.txt:46]:
```suggestion
   Files with these names must conform to a specific schema for their component type. 
```

**Review by mongodben — APPROVED** [review]
lgtm!

**Pull request merged**